blocks/bricks/sequences.py

```diff
@@ -107,11 +112,15 @@ class MLP(Sequence, Initializable, Feedforward):
 
     """
     @lazy(allocation=['dims'])
-    def __init__(self, activations, dims, **kwargs):
+    def __init__(self, activations, dims, prototype=None, **kwargs):
         self.activations = activations
-
-        self.linear_transformations = [Linear(name='linear_{}'.format(i))
-                                       for i in range(len(activations))]
+        self.prototype = Linear() if prototype is None else prototype
+        self.linear_transformations = []
+        for i in range(len(activations)):
+            linear = copy.deepcopy(self.prototype)
```

Comment:
Right, I didn't think of asking users to use `partial`, I agree that would be the better option. Should probably do the same for `parallel` bricks then though. We can make an issue for that if you don't feel like making it part of this PR.